wagtail/contrib/modeladmin/middleware.py

```diff
@@ -0,0 +1,43 @@
+from django.utils.six.moves.urllib.parse import urlparse
+from django.http import HttpResponseRedirect
+from django.core.urlresolvers import resolve, reverse, Resolver404
+
+
+class ModelAdminMiddleware(object):
```

Comment:
Can we remove this middleware? It looks like something that really should be solved at the view level.